src/quetoo.h

```diff
@@ -63,20 +63,40 @@
 	#define lengthof(x) (sizeof(x) / sizeof(x[0]))
 #endif
 
-typedef float vec_t;
-typedef vec_t vec2_t[2];
-typedef vec_t vec3_t[3];
-typedef vec_t vec4_t[4];
-
-typedef double dvec_t;
-typedef dvec_t dvec2_t[2];
-typedef dvec_t dvec3_t[3];
-typedef dvec_t dvec4_t[4];
-
-typedef uint16_t u16vec_t;
-typedef u16vec_t u16vec2_t[2];
-typedef u16vec_t u16vec3_t[3];
-typedef u16vec_t u16vec4_t[4];
+#define VECTOR_TYPENAME(name) name ## _t
+#define VECTOR_TYPENAME_N(name, n) name ## n ## _t[n]
+#define VECTOR_TYPE(type, typename) \
+	typedef type VECTOR_TYPENAME(typename); \
+	typedef VECTOR_TYPENAME(typename) VECTOR_TYPENAME_N(typename, 2); \
+	typedef VECTOR_TYPENAME(typename) VECTOR_TYPENAME_N(typename, 3); \
+	typedef VECTOR_TYPENAME(typename) VECTOR_TYPENAME_N(typename, 4);
+
+VECTOR_TYPE(float, vec);
+VECTOR_TYPE(double, dvec);
+VECTOR_TYPE(_Bool, bvec);
```

Comment:
Do we use bvec anywhere? If not, I'd recommend we lose this so that we don't slip up and think it's 'byte vector'. Just my $0.02. Otherwise, this is nice.